.github/workflows/lint.yml

```diff
@@ -0,0 +1,20 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
```

Comment:
I'm not sure about using the `master` ref here.

Also, could you give a context about why `ludeeus/action-shellcheck` over https://github.com/koalaman/shellcheck/wiki/GitHub-Actions?

Reply:
I took the action-shellcheck block from another repo, but I now can't find the repo I copied from... I'll switch to the canonical one.

Reply:
Hmm, the shellcheck action you linked to is for differential running. I don't think that's what we want here, I'd prefer all files in `docker` were always scanned. Also we only want to scan `./docker` as we have submodules with shell scripts which shouldn't be shellchecked as part of docker-builds CI. The action you linked to does not allow restricting paths (as far as I can see).

Reply:
Actually, we don't want to run all repository files linting on a PR. The linter should lint only PR-affected changes.
The reason is that PR should have a single responsibility, but blocking it on unrelated issues is not a PR concern.

Reply:
could be initially copied from here https://github.com/temporalio/cli/blob/1f078f2af258b5f6b1b14c7ef52d87a7f0e32905/.github/workflows/ci.yml#L33